crates/driver/src/infra/api/quote/dto/quote.rs

```diff
@@ -0,0 +1,110 @@
+use {
+    crate::{
+        domain::{competition, eth},
+        util::serialize,
+    },
+    serde::Serialize,
+    serde_with::serde_as,
+};
+
+impl Quote {
+    pub fn from_domain(quote: &competition::quote::Quote) -> Self {
+        Self {
+            sell_token: quote.sell.token.into(),
+            buy_token: quote.buy.token.into(),
+            sell_amount: quote.sell.amount,
+            buy_amount: quote.buy.amount,
+            interactions: quote
+                .interactions
+                .iter()
+                .map(|interaction| match interaction {
+                    competition::solution::Interaction::Custom(interaction) => {
+                        Interaction::Custom(CustomInteraction {
+                            internalize: interaction.internalize,
+                            target: interaction.target.into(),
+                            value: interaction.value.into(),
+                            call_data: interaction.call_data.clone(),
+                            inputs: interaction
+                                .inputs
+                                .iter()
+                                .map(|asset| Asset {
+                                    token: asset.token.into(),
+                                    amount: asset.amount,
+                                })
+                                .collect(),
+                            outputs: interaction
+                                .outputs
+                                .iter()
+                                .map(|asset| Asset {
+                                    token: asset.token.into(),
+                                    amount: asset.amount,
+                                })
+                                .collect(),
+                        })
+                    }
+                    competition::solution::Interaction::Liquidity(_) => todo!(),
+                })
+                .collect(),
+        }
+    }
+}
+
+#[derive(Debug, Serialize)]
+#[serde_as]
+#[serde(rename_all = "camelCase")]
+pub struct Quote {
+    // TODO I think this might be a mistake. Instead of these four fields, how about this just
+    // returns a single amount field which is a U256? That should be enough, right? It would
+    // also simplify some of the other code which is nice.
+    sell_token: eth::H160,
+    buy_token: eth::H160,
+    #[serde_as(as = "serialize::U256")]
+    sell_amount: eth::U256,
+    #[serde_as(as = "serialize::U256")]
+    buy_amount: eth::U256,
```

Comment:
I believe one argument in the past for 4 values was improved debuggability. Instead of stitching together request and response to know all the details of a quote you get all the data in the response.

Reply:
I prefer not returning duplicate data back. IMO that just makes code more error prone rather than less. It makes logging slightly better but I don't like designing the types around that.